runtime/compiler/p/codegen/J9TreeEvaluator.cpp

```diff
@@ -11923,6 +11924,41 @@ TR::Register *J9::Power::TreeEvaluator::directCallEvaluator(TR::Node *node, TR::
       TR::Node::recreate(node, TR::vadd);
       return vaddEvaluator(node, cg);
       }
+   else if (callee->getRecognizedMethod() == TR::jdk_internal_vm_vector_VectorSupport_binaryOp &&
+       node->getOpCodeValue() == TR::vcall) // was vectorized
+      {
+      TR::DataType dataType = TR_VectorAPIExpansion::getDataTypeFromClassNode(node->getChild(2));
+      bool supported = true;
+      if (dataType != TR::Float)
+         supported = false;
+      if (!node->getChild(0)->getOpCode().isLoadConst() ||
+          node->getChild(0)->getInt() != TR_VectorAPIExpansion::VECTOR_OP_ADD)
+         supported = false;
+      if (!node->getChild(3)->getOpCode().isLoadConst() ||
+          node->getChild(3)->getInt() != 4)
+         supported = false;
+
+      TR_ASSERT_FATAL(supported, "Vector API opcode, type, and number of lanes should be supported\n");
```

Comment:
Yeah, I understand. I should've changed the code so that vcall is always a subject to the TR_UseVcall check. I think we can make the code more clear when we actually start using it. Then will know better how to add new opcodes and evaluators incrementally.